runelite-client/src/main/java/net/runelite/client/plugins/herbiboars/HerbiboarTrail.java

```diff
@@ -82,28 +79,28 @@
 			trailIds.add(trail.trailId);
 			trailIds.add(trail.trailId + 1);
 
-			allObjectLocs.addAll(Arrays.asList(trail.getObjectLocs(1)));
-			allObjectLocs.addAll(Arrays.asList(trail.getObjectLocs(2)));
+			allObjectLocs.add(trail.getObjectLoc(1));
+			allObjectLocs.add(trail.getObjectLoc(2));
 		}
 	}
 
-	public WorldPoint[] getObjectLocs(int varbitValue)
+	public WorldPoint getObjectLoc(int varbitValue)
 	{
 		switch (varbitValue)
 		{
 			case 1:
-				return new WorldPoint[]{objectLoc1, objectLoc3};
+				return objectLoc1;
 			case 2:
-				return new WorldPoint[]{objectLoc2, objectLoc4};
+				return objectLoc2;
 			case 0:
 			default:
-				return new WorldPoint[]{};
+				return null;
 		}
 	}
 
 	@Override
 	public String toString()
 	{
-		return String.format("trailId=%s obj1=%s obj2=%s obj3=%s obj4=%s", trailId, objectLoc1, objectLoc2, objectLoc3, objectLoc4);
+		return String.format("trailId=%s obj1=%s obj2=%s", trailId, objectLoc1, objectLoc2);
```

Comment:
Just use the `@ToString` annotation (or w/e is its name) from lombok

Reply:
Agreed